### PR TITLE
Remove Trace class

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves https://github.com/open-telemetry/opentelemetry-java/pull/218#issuecomment-489797322

Signed-off-by: Pavol Loffay <ploffay@redhat.com>

### CODE REVIEW

File: api/src/test/java/io/opentelemetry/OpenTelemetryTest.java [hunk 1]
```diff
@@ -50,7 +50,7 @@ public void before() {
 
   @Test
   public void testDefault() {
-    assertThat(OpenTelemetry.getTracer()).isEqualTo(Trace.getTracer());
+    assertThat(OpenTelemetry.getTracer()).isInstanceOf(NoopTrace.newNoopTracer().getClass());
```
Comment (carlosalberto) [inline api/src/test/java/io/opentelemetry/OpenTelemetryTest.java]: isInstanceOf(NoopTrace.class)?

File: opentracing-shim/src/main/java/io/opentelemetry/opentracingshim/TracerShim.java [hunk 2]
```diff
@@ -37,7 +37,7 @@
    * @since 0.1.0
    */
   public TracerShim() {
-    this(Trace.getTracer());
+    this(NoopTrace.newNoopTracer());
```
Comment (carlosalberto) [inline opentracing-shim/src/main/java/io/opentelemetry/opentracingshim/TracerShim.java]: Actually I'd like this one to call `OpenTelemetry.getTracer()` ;)

File: api/src/test/java/io/opentelemetry/OpenTelemetryTest.java (hunk 1 above)
Comment (pavolloffay) [reply api/src/test/java/io/opentelemetry/OpenTelemetryTest.java]: No, the tracer impl is private member of `NoopTrace`

File: opentracing-shim/src/main/java/io/opentelemetry/opentracingshim/TracerShim.java (hunk 2 above)
Comment (pavolloffay) [reply opentracing-shim/src/main/java/io/opentelemetry/opentracingshim/TracerShim.java]: +1 I will update it

**Pull request merged**